pkgs/applications/misc/archivy/default.nix

```
@@ -49,21 +49,21 @@ buildPythonApplication rec {
     hash = "sha256-o5dVJDbdKgo6hMMU9mKzoouSgVWl7xSAp+Aq61VcfeU=";
   };
 
-  # Relax some dependencies
-  postPatch = ''
-    substituteInPlace requirements.txt \
-      --replace 'WTForms ==' 'WTForms >=' \
-      --replace 'attrs == 20.2.0' 'attrs' \
-      --replace 'elasticsearch ==' 'elasticsearch >=' \
-      --replace 'python_dotenv ==' 'python_dotenv >=' \
-      --replace 'python_frontmatter == 0.5.0' 'python_frontmatter' \
-      --replace 'requests ==' 'requests >=' \
-      --replace 'validators ==' 'validators >=' \
-      --replace 'flask-login == ' 'flask-login >= ' \
-      --replace 'tinydb ==' 'tinydb >=' \
-      --replace 'Flask_WTF == 0.14.3' 'Flask_WTF' \
-      --replace 'Flask ==' 'Flask >='
-  '';
+  nativeBuildInputs = [ fixPythonDepsHook ];
+
+  relaxPythonDeps = [
+    "WTForms"
+    "attrs"
+    "elasticsearch"
+    "python_dotenv"
+    "python_frontmatter"
+    "requests"
+    "validators"
+    "tinydb"
+    "flask-login"
+    "Flask_WTF"
+    "Flask"
```

```suggestion
    "*"
```

At this point we might as well just do this.

---

I am open for a suggestion on how to do this. Should we treat `*` as a special case? Should we have a regex/glob parser so we allow people doing things like `Flask.*`?

---

I am thinking a completely different approach too. Maybe we want to always relax version restrictions, so we could instead just have the hook strip all version restrictions and keep the `removePythonDeps` for the cases where we want to remove a dependency for some reason (removing linters for example).

---

Another option: `relaxPythonDeps = true;` So if it is a boolean, we relax/remove all dependencies, otherwise we follow the list.

---

Done (for now).

---

> Another option: `relaxPythonDeps = true;` So if it is a boolean, we relax/remove all dependencies, otherwise we follow the list.

Done. Still open for better suggestion, though.

---

> I am thinking a completely different approach too. Maybe we want to always relax version restrictions, so we could instead just have the hook strip all version restrictions and keep the `removePythonDeps` for the cases where we want to remove a dependency for some reason (removing linters for example).

That sounds interesting but definitely needs documentation.



> Another option: `relaxPythonDeps = true;` So if it is a boolean, we relax/remove all dependencies, otherwise we follow the list.

Even better.